src/eigen.jl

```diff
@@ -79,8 +79,9 @@ end
     H1 = Hessenberg{typeof(zero(eltype(S.data))),typeof(T),typeof(S.data),typeof(tau),typeof(false)}(T, 'L', S.data, tau, false)
     A = S.data
     H = SymTridiagonal(zeros(eltype(E), n), E)
-    trisol = eigen!(H)
-
+    shift = norm(H)
```

Comment:
You should probably do the shift inside the tridiagonal eigensolver, since the error happens at that level.  (Also, the norm is cheaper to compute for a tridiagonal matrix.)